pkg/river/internal/value/value.go

```diff
@@ -25,9 +25,11 @@ var (
 	goCapsule         = reflect.TypeOf((*Capsule)(nil)).Elem()
 	goDuration        = reflect.TypeOf((time.Duration)(0))
 	goDurationPtr     = reflect.TypeOf((*time.Duration)(nil))
-	goRiverDecoder    = reflect.TypeOf((*Unmarshaler)(nil)).Elem()
-	goRawRiverFunc    = reflect.TypeOf((RawFunction)(nil))
-	goRiverValue      = reflect.TypeOf(Null)
+	// goRiverDefaulter  = reflect.TypeOf((*Defaulter)(nil)).Elem()
```

Comment:
I suspect updates to decode.go will be needed and these commented out lines will become useful. I wanted to start with this prototype before going too far into it.

Reply:
Yeah, Defaulter/Validator should be called roughly in the same place as where Unmarshaler is used.